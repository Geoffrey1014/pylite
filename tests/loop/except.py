class Generator:
    def __init__(self, start, end):
        self.start = start
        self.end = end
        self.current = start

    def __iter__(self):
        return self

    def __next__(self):
        if self.current == (self.start + self.end) / 2:
            0 / 0
        current = self.current 
        self.current = self.current + 1
        return current

g = Generator(0, 10)
for i in g:
    print(i)
