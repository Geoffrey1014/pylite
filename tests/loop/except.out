0
1
2
3
4
ZeroDivisionError:
  At except.py, line 12, in __next__()
  At except.py, line 18, in __init__()
